### PR TITLE
feat(changelog): allow releases to lack a body if it contains a name

### PR DESCRIPTION
**Pull request by HenrikPoulsen**
## Changes

* Prevents Renovate from filtering out release notes that doesn't have a release body, if it has a release name.
* The PR template also includes the release name in the Release Notes section if one is present.

## Context
See #18689 for additional context.
I have some dependencies that frequently includes useful information in the release name.
Renovate does not show this information in the PR release notes section and it would be very useful to me (and likely others) to have this additional information available when it exists.

See PR [here](https://github.com/HenrikPoulsen/renovate-test/pull/2) for what the change would look like. Version v4.2.0 has the release name `Add check-latest input and bug fixes` and is added to the release notes section for that version.

One observation I've made is that some repos seem to set the release name to the same as the tag name. So with this change the release ends up with `v.4.3.0: v4.3.0` for those releases. Do we expect that to be an issue that needs to be prevented?

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
Do you have any public examples of releases with a name but no body?

File: lib/workers/repository/update/pr/changelog/release-notes.ts [hunk 1]
```diff
@@ -171,7 +171,7 @@ async function releaseNotesResult(
   }
   // set body for release notes
   releaseNotes.body = massageBody(releaseNotes.body, baseUrl);
-  if (releaseNotes.body.length) {
+  if (releaseNotes.body.length || releaseNotes.name?.length) {
```
Comment (viceice) [inline lib/workers/repository/update/pr/changelog/release-notes.ts:174]: We need to filter when release name equals version, which would duplicate the version and would happen for sure

**Review by viceice — CHANGES_REQUESTED** [review]

**Comment by HenrikPoulsen** [comment]
> Do you have any public examples of releases with a name but no body?

Sadly not of the top of my head. We have several internal packages that frequently only uses the name in the release (mostly for very small fixes)

File: lib/workers/repository/update/pr/changelog/release-notes.ts (hunk 1 above)
Comment (rarkins) [reply lib/workers/repository/update/pr/changelog/release-notes.ts:174]: Yeah, strip out version prefix, trim it, and then include it only if text remains

**Review by viceice — APPROVED** [review]

**Pull request merged**